src/main/java/com/launchdarkly/client/StreamProcessor.java

```diff
@@ -171,4 +195,21 @@ int version() {
     }
 
   }
+
+  private final class HeartbeatDetector implements Runnable {
+
+    @Override
+    public void run() {
+      DateTime fiveMinutesAgo = DateTime.now().minusSeconds(DEAD_CONNECTION_INTERVAL_SECONDS);
+      if (lastHeartbeat.isBefore(fiveMinutesAgo) && es.getState() == ReadyState.OPEN) {
+        try {
+          logger.info("Stream stopped receiving heartbeats- reconnecting.");
+          es.close();
+          start();
```

Comment:
What if we're actually failing to close the previous EventSource? If that's happening every time (for some unknown reason) then we'll be creating a new ES instance every few minutes.